### DISCUSSION TITLE
请问这个区域怎么关闭？

### DISCUSSION THREAD

**Discussion by ghost** [umijs/umi#11168] [Q&A]
getInitialState我看只能去掉名字，我看链接文档上没有说明，或者怎么去掉这整个区域
![image](https://github.com/umijs/umi/assets/37528714/155f39d5-4cf4-498d-bdbf-7b016e0d41b7)

**Comment by fz6m** [comment]
layout 里的选项，在 `export const layout = () => ` 返回值里配置，详见 [pro layout](https://procomponents.ant.design/components/layout) 文档。

**Reply by ghost** [reply]
![image](https://github.com/umijs/umi/assets/37528714/c1084a63-5268-44a1-a952-63fb13590e3d)

app.ts代码如下

export async function getInitialState() {
  return { name: '@umijs/max' }
}

export const layout = () => {
  return {
    logo: 'https://img.alicdn.com/tfs/TB1YHEpwUT1gK0jSZFhXXaAtVXa-28-27.svg',
    layout: 'side',
    menuFooterRender: false,
    avatarProps: {
      src: null,
      size: 'small'
    },
    menu: {
      locale: false
    }
  }
}

**Reply by ghost** [reply]
![image](https://github.com/umijs/umi/assets/37528714/313d8fd3-1e05-4cc9-9cb8-c06dd2e14a46)

layout: 'top' 是这样

**Reply by ghost** [reply]
![image](https://github.com/umijs/umi/assets/37528714/c8900938-e20f-4c9f-9058-2a26884904bd)

事实上getInitialState这样返回，那个有颜色小头像不显示了，但是元素还在

app.ts代码如下
// 运行时配置

// 全局初始化数据配置，用于 Layout 用户信息和权限初始化
// 更多信息见文档：https://umijs.org/docs/api/runtime-config#getinitialstate
export async function getInitialState() {
  return { name: '', avatar: { src: null } }
}

export const layout = () => {
  return {
    logo: 'https://img.alicdn.com/tfs/TB1YHEpwUT1gK0jSZFhXXaAtVXa-28-27.svg',
    layout: 'side',
    menuFooterRender: false,
    avatarProps: {
      src: null,
      size: 'small'
    },
    menu: {
      locale: false
    }
  }
}

**Reply by ghost** [reply]
@fz6m

**Reply by ghost** [reply]
"@umijs/max": "^4.0.69", "@ant-design/pro-components": "^2.4.16", node: v16.20.0

**Reply by fz6m** [reply]
隐藏方法：

```ts
  rightContentRender: false
```

**Reply by ghost** [reply]
感谢，我尝试了一下，rightContentRender和rightRender在app.tsx都是可以的，原先的是运行时文件是app.ts命名格式，这个时候rightRender: false隐藏还会剩下一个头像，rightContentRender: false是可以的
